core/src/main/java/hudson/model/Queue.java

```diff
@@ -2079,6 +2079,16 @@ default boolean isConcurrentBuild() {
          */
         @NonNull SubTask getParent();
 
+        /**
+         * An umbrella executable (such as a {@link Run}) of which this is one part.
+         * If {@link #getParent} has a distinct {@link SubTask#getOwnerTask},
+         * then it should be the case that {@code getParentExecutable().getParent() == getParent().getOwnerTask()}.
+         * @return a <em>distinct</em> executable (unlike the default of {@link SubTask#getOwnerTask}!); or null if this executable was already at top level
```

Comment:
What motivated this difference?

Reply:
Just general sanity—it is pretty confusing for a method named to look for the parent of some object to return the same object. I did not care to propagate the weird design.

Reply:
```suggestion
         * @return a <em>distinct</em> executable (never {@code this}, unlike the default of {@link SubTask#getOwnerTask}!); or null if this executable was already at top level
```